src/Functions/splitByChar.cpp

```diff
@@ -98,7 +99,7 @@ class SplitByCharImpl
                    return false;
         }
 
-        pos = reinterpret_cast<Pos>(memchr(pos, separator, end - pos));
+        pos = reinterpret_cast<Pos>(memchrSmallAllowOverflow15(pos, separator, end - pos));
```

Comment:
Does that not only work for inputs with >= 16 bytes length? Otherwise, we'll scan garbage behind the end of the value.

Reply:
Equally important: Is the standard library `memchr` not already optimized to death (using SIMD) for sufficiently large inputs?

Reply:
> Does that not only work for inputs with >= 16 bytes length? Otherwise, we'll scan garbage behind the end of the value.

That's ok. `memchrSmallAllowOverflow15Impl` already considered this situation.

Reply:
> Equally important: Is the standard library `memchr` not already optimized to death (using SIMD) for sufficiently large inputs?

Seems not according to test result below.  

```
with 'Many years later as he faced the firing squad, Colonel Aureliano Buendia was to remember that distant afternoon when his father took him to discover ice.' as s select splitByChar(' ', materialize(s)) as w from numbers(1000000) format Null; 
```

use memchr
```
0 rows in set. Elapsed: 0.413 sec. Processed 1.00 million rows, 8.00 MB (2.42 million rows/s., 19.36 MB/s.)
Peak memory usage: 62.90 MiB. 
```

use memchrSmallAllowOverflow15
```
0 rows in set. Elapsed: 0.355 sec. Processed 1.00 million rows, 8.00 MB (2.81 million rows/s., 22.52 MB/s.)
Peak memory usage: 62.90 MiB. 
```